### PR TITLE
Update Space-0 Token To Have a Unit

### PR DESCRIPTION
**Pull request by benwolfram**
### WHY are these changes introduced?

Using a unitless value (aka `0`) inside of a `calc` causes `calc` to error, thus breaking the CSS rule you are trying to use. Here, I am adding a unit to the value to fix this problem. Per [this design tokens format module](https://design-tokens.github.io/community-group/format/#dimension), values should include units, including 0.

### WHAT is this pull request doing?

This is allowing for the `space-0` token to be used inside of a `calc`.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

I built the tokens package and saw the correct value was outputted in the build file.

```
--p-space-0:0rem;
```

Documentation is updated

image

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by aaronccasanova** [comment]
LGTM 👍 Mind adding a changeset entry with `yarn changeset` and cutting a snapshot?

**Comment by benwolfram** [comment]
Should have that now I think @aaronccasanova

**Comment by aaronccasanova** [comment]
NPM distribution and Storybook looks good 👍 

image

image

File: .changeset/clever-sheep-sip.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-tokens': patch
+---
+
+Adding a unit to the space-0 token
```
Comment (aaronccasanova) [inline .changeset/clever-sheep-sip.md]:
```suggestion
Added a unit to the `space-0` token
```

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**